### PR TITLE
[Progress] Add user select dropdown

### PR DESCRIPTION
**Pull request by lfryemason**
## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/UnitSelectorV2.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      const newScriptId = e.target.value;
+      setScriptId(newScriptId);
+      loadUnitProgress(newScriptId, sectionId);
+
+      this.recordEvent('change_script_v2', {
+        old_script_id: scriptId,
+        new_script_id: newScriptId,
+      });
+
+      analyticsReporter.sendEvent(EVENTS.PROGRESS_CHANGE_UNIT, {
```
Comment (lfryemason) [inline apps/src/templates/UnitSelectorV2.jsx]: Change to new event

File: apps/src/templates/UnitSelectorV2.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      });
+
+      analyticsReporter.sendEvent(EVENTS.PROGRESS_V2_CHANGE_UNIT, {
+        sectionId: sectionId,
+        oldUnitId: scriptId,
+        unitId: newScriptId,
+      });
+    },
+    [scriptId, setScriptId, sectionId]
+  );
```
Comment (etaderhold) [inline apps/src/templates/UnitSelectorV2.jsx]: The script/unit synonym makes this a bit hard to follow. Could you perhaps replace `script` with `unit` wherever it's easy? Like, `setScriptId` is probably used in a bunch of places so that can be a refactor for another time, but I'd love to see us standardize on `unit` for new local variables, event names, etc.

**Review by etaderhold — APPROVED** [review]

File: apps/src/templates/UnitSelectorV2.jsx (hunk 2 above)
Comment (lfryemason) [reply apps/src/templates/UnitSelectorV2.jsx]: the scriptId is used in 17 files and `setScriptId` is used in 16. I think I'd prefer to put that into a different PR. I did however change the newScriptId and also set `unitId = scriptId` in the file.
